osfoffline/database_manager/models.py

```diff
@@ -100,10 +101,9 @@ def path(self):
         # +os.path.sep+ instead of os.path.join: http://stackoverflow.com/a/14504695
 
         if self.parent:
-            return os.path.join(self.parent.path, 'Components', self.title)
+            return os.path.join(self.parent.path, 'Components', make_folder_name(self.title, self.osf_id))
```

Comment:
According to your own function definition, the second argument is a kwarg, not positional- update to match.